snippets/floated-label.md

```diff
@@ -0,0 +1,82 @@
+### Floated label
+
+One of most common used pattern is input plus label. Using CSS to create floated label animation is achievable. And actually, it's simple with `:placeholder-shown` property.
+
+#### HTML
+
+```html
+<div class="floated-label-container">
+  <label for="input" class="input-label">
+    <input placeholder="&nbsp;" id="input">
+    <span class="label">Label</span>
+  </label>
+</div>
+```
+
+#### CSS
+
+```css
+.floated-label-container {
+  font-family: Arial, Helvetica, sans-serif;
+  height: 100px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.input-label input {
+  border: 0;
+  border-bottom: 2px solid #c8ccd4;
+  padding: 8px 0;
+  width: 250px;
+  background: #f5f6f9;
+}
+
+.input-label input:hover {
+  background: #f3f3f3;
+}
+
+.input-label input:focus {
+  background: none;
+  outline: none;
+}
+
+.input-label > .label {
+  position: absolute;
+  margin-top: -32px;
+  color: #9e9e9e;
+  transition: transform 0.1s ease-out;
+}
+
+.input-label > input:not(:placeholder-shown) + .label,
+.input-label > input:focus + .label {
+  color: grey;
+  transform: translate(-4px, -22px) scale(0.75);
```

Comment:
We can remove the x-axis translation if we add `transform-origin: left` to the `> .label` selector.

Reply:
Yeah, it reduce translate X axis transform for the element. 👍